### PR TITLE
UW-425

### PR DESCRIPTION
**Pull request by WeirAE**
**Description**
Add documentation for main page. Replaces and combines individual index pages.
Preview docs are [here](https://unified-workflow--366.org.readthedocs.build/en/366/). 

**Type**
- [x] Documentation

**Impact**
- [x] This is a non-breaking change (existing functionality continues to work as expected)

**Checklist**

Affirm:

- [x] I have added myself and any co-authors to the PR's _Assignees_ list.
- [x] I have reviewed the documentation and have made any updates necessitated by this change.

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -1,14 +1,100 @@
...
+   ├── docs 
+   │     └── sections
+   │           ├── contributor_guide
+   │           └── user_guide
+   └── src
+         └── (uwtools)
+    	         ├── api
+    	         ├── apps 
+    	         ├── config
+               │     └── formats
```
Comment (NaureenBharwaniNOAA) [inline docs/index.rst:41]: `formats` got a bit off center in the documentation on the post.

File: docs/index.rst [hunk 2]
```diff
@@ -1,14 +1,100 @@
...
+               │     ├── api
+               │     ├── config
+               │     │     └── formats
+               │     ├── drivers
+               │     ├── files
+               │     │     ├── gateway
+               │     │     ├── interface
+               │     │     └── model
+               │     ├── fixtures
+               │     │     └── files
```
Comment (NaureenBharwaniNOAA) [inline docs/index.rst:58]: I would take a look again at this table. Maybe removing the `|` to the left of nested items will help this format better.

Confirming I am looking at this documentation: https://unified-workflow--366.org.readthedocs.build/en/366/#

**Review by NaureenBharwaniNOAA — COMMENTED** [review]
A couple updates to the documentations formatting itself on the website page. Other than that it sounds great!

File: docs/index.rst [hunk 3]
```diff
@@ -1,14 +1,100 @@
...
+
+A schematic of the above is shown in :numref:`Schematic_Layout_Unified_Workflow_Framework` below.
```
[the remark this answers is not in the document]
Comment (WeirAE) [reply docs/index.rst:11]: Yeah, 'above' text should probably only be used if formatted as a caption - though, this is still the original text and diagram, I thought about it and I think we should incorporate text and diagrams from your poster and Christina's presentation instead

File: docs/index.rst [hunk 4]
```diff
@@ -1,14 +1,100 @@
...
+
+.. _Schematic_Layout_Unified_Workflow_Framework:
+	
+.. figure:: /static/figure/Schematic_Layout_Unified_Workflow_Framework.png
```
[the remark this answers is not in the document]
Comment (WeirAE) [reply docs/index.rst:15]: Yeah, the image in this branch is stored under `old_docs`, I'm fussing at replacing it

File: docs/index.rst (hunk 1 above)
Comment (WeirAE) [reply docs/index.rst:41]: aha, found the problem - somehow there was a half-tab on some lines instead of 4 spaces, fixed it!

**Comment by WeirAE** [comment]
Closing out to switch to a different task

**Pull request closed without merging**